From 3a76ebe0b3f8375007e4705d910a4301b59af634 Mon Sep 17 00:00:00 2001
From: Lu Yahan <yahan@iscas.ac.cn>
Date: Mon, 13 Sep 2021 17:12:30 +0800
Subject: [PATCH] [riscv64] Add force constant pool into
 PrepareForDeoptimizationExits

Change-Id: I9b65bdc772b00177e48547823c3a3bea79f8b875
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/3156589
Auto-Submit: Yahan Lu <yahan@iscas.ac.cn>
Commit-Queue: Ji Qiu <qiuji@iscas.ac.cn>
Reviewed-by: Ji Qiu <qiuji@iscas.ac.cn>
Cr-Commit-Position: refs/heads/main@{#76806}
---
 src/compiler/backend/riscv64/code-generator-riscv64.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/compiler/backend/riscv64/code-generator-riscv64.cc b/src/compiler/backend/riscv64/code-generator-riscv64.cc
index 02ce2b67134..c647f6714c8 100644
--- a/src/compiler/backend/riscv64/code-generator-riscv64.cc
+++ b/src/compiler/backend/riscv64/code-generator-riscv64.cc
@@ -3062,6 +3062,7 @@ void CodeGenerator::FinishCode() { __ ForceConstantPoolEmissionWithoutJump(); }
 
 void CodeGenerator::PrepareForDeoptimizationExits(
     ZoneDeque<DeoptimizationExit*>* exits) {
+  __ ForceConstantPoolEmissionWithoutJump();
   int total_size = 0;
   for (DeoptimizationExit* exit : deoptimization_exits_) {
     total_size += (exit->kind() == DeoptimizeKind::kLazy)
-- 
2.35.1

